graviti/portex/field.py

```diff
@@ -85,6 +100,152 @@ def rename(self, old_name: str, new_name: str) -> None:
         self._keys.__setitem__(self._keys.index(old_name), new_name)
 
 
+UnionFields = Union[ImmutableFields, MutableFields]
+
+
+class ConnectedFields(MutableMapping[str, PortexType]):
+    """Fields composed of ImmutableFields and MutableFields.
+
+    Raises:
+        ValueError: When there as repeated field names.
+
+    Arguments:
+        fields: The ImmutableFields and MutableFields.
+
+    """
+
+    def __init__(self, multi_fields: Iterable[UnionFields]) -> None:
+        multi_fields = list(multi_fields)
+        field_names: Set[str] = set()
+        for fields in multi_fields:
+            names = set(fields.keys())
+            repeated_keys = field_names & names
+            if repeated_keys:
+                raise ValueError(f"Repeated field names '{repeated_keys}'")
+            field_names = field_names.union(names)
+
+        chained_fields = ChainMap(*multi_fields)
+        self._chained_fields: MutableMapping[str, PortexType] = chained_fields
+        self._multi_fields: List[UnionFields] = chained_fields.maps  # type: ignore[assignment]
+
+    def __len__(self) -> int:
+        return sum(map(len, self._multi_fields))
+
+    def __getitem__(self, key: Union[int, str]) -> PortexType:
+        if isinstance(key, int):
+            i, j = self._get_indices(key)
+            return self._multi_fields[i][j]
+
+        return self._chained_fields[key]
+
+    def __setitem__(self, key: Union[int, str], value: PortexType) -> None:
+        if isinstance(key, int):
+            i, key = self._get_indices(key)
+            fields = self._multi_fields[i]
+        else:
+            fields = (
+                self._get_fields(key) if key in self._chained_fields else self._multi_fields[-1]
+            )
+
+        fields[key] = value  # type: ignore[index]
```

Comment:
if fields is immutable? this line will raise `AttributeError`? user may not know what happens?